pages/goals.tsx

```diff
@@ -6,132 +6,84 @@ import {
+    FETCH_USER_GOALS,
+    {
+      variables: {
+        userId: user.uid,
+      },
 
-  // sample data for Goals Sections, eventually pull from hasura and map to this data structure
+      onCompleted: (data: FetchUserGoalsDataResponse) => {
+        setUserGoals(data.user_goals);
+      },
+    }
+  );
 
   const tailwindIconSize = "h-5 w-5";
 
-  const GoalsSections = [
+  // create goal sections here, map through below to render
+
+  const goalsSections: GoalsSectionType[] = [
     {
       header: {
         sectionName: "Current Goals",
-        // array of heroicon components
         manageIcons: [
           <ArchiveIcon className={tailwindIconSize} />,
-          <TrashIcon className={tailwindIconSize} />,
           <CheckCircleIcon className={tailwindIconSize} />,
         ],
       },
-      rows: [
-        {
-          goalName: "Learn React Fundamentals",
-          createdDate: "2022-06-01",
-          targetDate: "2022-06-14",
-        },
-        {
-          goalName: "First 5 problems of LeetCode 75",
-          createdDate: "2022-06-01",
-          targetDate: "2022-06-07",
-        },
-        {
-          goalName: "Lorem Ipsum Dolor Sit Amet",
-          createdDate: "2022-06-07",
-          targetDate: "2022-07-01",
-        },
-      ],
+      rows: userGoals.filter((goal) => goal.isActive),
     },
-
     {
       header: {
         sectionName: "Completed Goals",
-        // array of heroicon components
         manageIcons: [
           <ArchiveIcon className={tailwindIconSize} />,
           <TrashIcon className={tailwindIconSize} />,
         ],
       },
-      rows: [
-        {
-          goalName: "Complete all levels of FlexBox Frog Game",
-          createdDate: "2022-06-01",
-          targetDate: "2022-06-08",
-        },
-        {
-          goalName: "Build Basic HTML page",
-          createdDate: "2022-06-01",
-          targetDate: "2022-06-06",
-        },
-        {
-          goalName: "Make first pull request",
-          createdDate: "2022-06-01",
-          targetDate: "2022-06-02",
-        },
-      ],
+      rows: userGoals.filter((goal) => goal.isComplete),
     },
-
     {
       header: {
         sectionName: "Archived Goals",
-        // array of heroicon components
-        manageIcons: [
-          <ArchiveIcon className={tailwindIconSize} />,
-          <TrashIcon className={tailwindIconSize} />,
-          <CheckCircleIcon className={tailwindIconSize} />,
-        ],
+        manageIcons: [<TrashIcon className={tailwindIconSize} />],
       },
-      rows: [
-        {
-          goalName: "Learn Kotlin",
-          createdDate: "2022-06-01",
-          targetDate: "2022-09-01",
-        },
-        {
-          goalName: "Learn SQL",
-          createdDate: "2022-06-01",
-          targetDate: "2022-09-01",
-        },
-      ],
+      rows: userGoals.filter((goal) => !goal.isComplete && !goal.isActive),
```

Comment:
so when a goal goes from complete to archived, we have to mark it as incomplete? 

would it make more sense to add a new column called isArchived? And then only show a goal as completed if `isComplete && !isArchived`